snuba/datasets/tags_column_processor.py

```diff
@@ -13,7 +16,54 @@
 )
 
 # A column name like "tags[url]"
-NESTED_COL_EXPR_RE = re.compile(r"^([a-zA-Z0-9_\.]+)\[([a-zA-Z0-9_\.:-]+)\]$")
+NESTED_COL_EXPR_RE = re.compile(
+    r"^([a-zA-Z0-9_\.]+)(?:\[((?:(?:[a-zA-Z0-9_\.\,:-]+)\s*)+)\])?$"
+)
+
+
+@dataclass(frozen=True)
```

Comment:
Everything in this file is kind of temporary, and due to the fact tags are still processed in the old query structure during the formatting phase. In the new AST this should become simpler. and happen earlier in the processing.